kitsune/wiki/tests/test_views.py

```diff
@@ -684,3 +685,170 @@ def test_it_works(self):
+                                                             req_doc_locale='bn-IN')
+        # As the Document is not translated into bn-IN and there is no locale of
+        # ACCEPT_LANUGAE header which the document to, But there is a custom wiki fallback
+        # mapping available for bn-IN. the fallback locale is bn-BD. So the user should
+        # see the document in bn-BD.
+        en_content = 'This article is in English'
+        trans_content = 'This article is translated into bn-BD'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+    def test_custom_locale_mapping_for_header_locale(self):
+        """
+        Test that configured locale fallbacks are used for header locales.
+
+        * The article is not translated in requested locale, header locales
+        * The header locales does not have any fallback locale or the article is not translated
+          into any of the fallback locale
+        * The requested locale have configured fallback locale but the article is not translated
+           to any of the fallback locale
+        * But one of the locale of ACCEPT_LANGUAGE header has configured fallback locale and the
+           article is translated into the fallback locale
+        *** Test that the Article is showing in the configured fallback locale
+        """
+
+        # While the requested locale also in custom wiki fallback locale but none of the
+        # fallback locale for requested locale is translated
+
+        # A Client with ca, pt-PT, ja language choices in ACCEPT_LANGUAGE Header
+        header = 'ca,pt-PT;q=0.7,ja;q=0.3,'
+        # Create a translated document in pt-BR and try to get ca locale version of the
+        # document by using above mentioned client.
+        doc_content = self.get_data_from_translated_document(header=header,
+                                                             create_doc_locale='pt-BR',
+                                                             req_doc_locale='ca')
+        # As the ca version of the document is not translated moreover the we have
+        # custom locale fallback mapping for ca locale, but the document is translated to none
+        # of the fallback locale of ca. But in the header, there is pt-PT locale and we have
+        # custom wiki locale mapping for pt-PT. the fallback locale is pt-BR. and the document is
+        # translated into pt-BR, so the user should see pt-BR vesion of the document.
+        en_content = 'This article is in English'
+        trans_content = 'This article is translated into pt-BR'
+        assert en_content not in doc_content
+        assert trans_content in doc_content
+
+        # While the Requested locale not in Custom Wiki fallback locale
+        # A Client with ca, bn-IN, ja language choices in ACCEPT_LANGUAGE Header
+        header = 'ar,bn-IN;q=0.7,ja;q=0.3,'
+        # Create a translated document into bn-BD and try to get the ar locale version
```

Comment:
Entire section rewrite:
# Create a document localized into bn-BD
# Attempt to resolve to the ar version of the document with the client defined before